catalyst/src/catalyst.js

```diff
@@ -0,0 +1,70 @@
+import {initializeSandbox} from './sandbox.js';
+
+export function initializeCatalyst(){
+  var renderRule = initializeSandbox('_general');
+  renderRule.sandboxes = [];
+
+  let renderRuleProxy = new Proxy(renderRule, {
+    get(target, name, receiver) {
+        let rv = Reflect.get(target, name, receiver);
+        if (!rv) {
+            target[name] = initializeSandbox(name); 
+            rv = Reflect.get(target, name, receiver);
+            renderRule.sandboxes.push(rv);
+        }
+        rv.lastAccessTime = new Date().getTime();
+        return rv;
+    }
+  });
+
+  const spaCleanupThreshold = 500;
```

Comment:
Occasionally, I think this threshold still causes me issues.  I don't recall exactly, but is this a performance thing or why not just consider every access fresh?

Reply:
There is a mechanism to provide your own check or to disable it. 
This is to compensate for the fact that we don't have a mechanism to query the eid or context id that is active for the code executing. If we did, we could have be an explicit and clean mechanism.

As an example, my build mechanism for vscode adds the following line to the bottom of the context code:
```
    rule.isActive = function(){
      return [...document.querySelector('html').classList].includes("evolv_web_${context.id}")
    };
```

This works very well for all circumstances, but requires the context code to know it's context id in the yaml.